src/sentry/web/frontend/organization_auth_settings.py

```diff
@@ -89,16 +119,18 @@ def handle_existing_provider(self, request, organization, auth_provider):
                 )
                 return self.redirect(next_uri)
 
-        form = AuthProviderSettingsForm(
-            data=request.POST if request.POST.get("op") == "settings" else None,
-            initial={
-                "require_link": not auth_provider.flags.allow_unlinked,
-                "default_role": organization.default_role,
-            },
-        )
+        form = auth_provider_settings_form(provider, auth_provider, organization, request)
 
         if form.is_valid():
             auth_provider.flags.allow_unlinked = not form.cleaned_data["require_link"]
+
```

Comment:
If all of this had a transaction wrapping it the problems noted above around partial persistence could be avoided.

Reply:
👍🏼

Reply:
looking at this, we decorate the entire handle method with `@transaction.atomic`  -- is an inner transaction still needed?